Decision records/05 - We use Details and List to describe index- and show-style route components.md

```diff
@@ -0,0 +1,43 @@
+# We use `Details` and `List` to describe index- and show-style route components
+
+## Date
+
+July 30, 2018
+
+## Contributors
+
+* Chris Sauvé
+* Mallory Allen
+* Tzvi Melamed
+* Ismail Syed
+* Gord Pearson
+* Matt Seccafien
+
+## Summary
+
+Many apps will have components that represent discrete routes in the application. These routes often represent traditional CRUD-style operations on a resource. In these cases, we use the terms "Details" (for a single resource) and "List" (for a list of resources) to name these routes.
+
+## History
+
+In Shopify Web, we initially named our "route" components after their Rails equivalents. For a resource like products, we had a `ProductIndex` component (for `/admin/products`), `ProductShow` (for `/admin/products/:id`), and `ProductCreate` (for `/admin/products/new`). These matched up the default naming style for Rails actions. 
+
+Eventually, there was so much duplication between `ProductShow` and `ProductCreate` that the two were merged into a single component, still named `ProductShow`.
+
+This same naming convention was copied to many other sections and apps that had similar CRUD-based resources.
+
+## Decision
+
+The `Show`/ `Create`/ `Index` postfix naming has one major benefit: it matches the Rails convention. This makes it simpler for Rails-knowledgeable developers to apply their existing mental model to the React application, and makes a conversion from a Rails codebase to a React one even simpler.
+
+However, there are numerous issues with this approach:
+
+* It does not necessarily line up with the way the page interacts with the GraphQL API (there is no CRUD in GraphQL)
+* It feels restrictive in a world outside of Rails, where there is no built-in assumptions about the structure of routes around a resource
+* It does not do a good job of indicating the responsibility of a merged show/ create component
+
+We considered the following alternative naming for the list/ detail pages:
+
+* `ResourceList` and `Resource`: `Resource` is overly generic and can easily conflict with other identifiers
+* `Resources` and `Resource`: same as the above, but with the added issue of overly similar names
+
+We feel that `ResourceList` and `ResourceDetails` provides the best balance as default naming. The names are sufficiently unique and descriptive, and align closely with how platforms like iOS refer to similar concepts ("master/ detail"). `ResourceList` does feel overly prescriptive on how the page should look, rather than its purpose, but we could not come up with better alternatives.
```

Comment:
Are we opposed to just duplicating the mobile naming?

Reply:
What’s the mobile naming?

Reply:
would this apply to things like apps which are perhaps less of a resource but more of an entry point to other resources or processes? e.g.`EmbeddedAppList` makes sense but `EmbeddedAppDetails` feels a little strange

Reply:
@gf3 is this because items in the list of apps actually launch the app, rather than opening a detail page for the app? (If so, the way I’d think about this is apps are still “resources”, but the list behavior is a bit different. The EmbeddedAppDetails convention would make sense for a page that offers info about the app, things like stats, permissions, uninstall, etc. I don’t think this naming convention would apply to the homepage for the app itself.)

Reply:
@gf3 in my view it would be fine to have `EmbeddedAppList`/ `EmbeddedApp`, as I agree it is not a resource in the way we think of other resources

Reply:
the mobile naming is `master` / `list` according to the doc